language/move-vm/runtime/src/runtime.rs

```diff
@@ -106,11 +106,21 @@ impl VMRuntime {
         // load the script, perform verification
         let (main, type_params) = self.loader.load_script(&script, &ty_args, data_store)?;
 
-        // build the arguments list for the main and check the arguments are of restricted types
-        let first_param_opt = main.parameters().0.get(0);
-        if first_param_opt.map_or(false, |sig| is_signer_reference(sig)) {
-            args.insert(0, Value::transaction_argument_signer_reference(sender))
+        // Build the arguments list for the main and check the arguments are of restricted types.
+        // Signers are built up from left-to-right. Not all signer arguments passed in may be used (e.g., if
+        // the script doesn't take a &signer, or if it isn't an admin script with one &signer).
+        let parameters = &main.parameters().0;
+        for (i, signer) in signers.into_iter().enumerate() {
```

Comment:
do we want to impose some correlation between signers passed in and expected?
I suppose we cannot because of the fact not all script take a signer, but I wonder....

Reply:
It's a good question, but I think not imposing a correlation is fine for a couple reasons:
* Historically we've allowed this as you mentioned, but also
* Using a prefix of the signers is at most as powerful as using the full list of signers, so from a privilege standpoint, not using all of the signers is fine. Essentially the VM is saying "these are all the account privileges you have" and the script is saying "ok I'll only take a couple" here which seems sensible to me.